File: backend/time-tracker-service/src/main/java/de/hnnsb/timetrackerservice/tasks/Task.java
package de.hnnsb.timetrackerservice.tasks;

import de.hnnsb.timetrackerservice.categories.Category;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;
import java.util.UUID;

@Entity
@Getter
@Setter
public class Task {
    @Id
    @GeneratedValue
    private UUID id;
    private String name;
    private String description;
    private Date startTime;
    private Date endTime;
    private String email;
    private int pauseTime;
    private Date pauseStart;
    @ManyToOne
    private Category category;
}

File: web-app-offline/src/lib/model/task.ts
import { Category } from "./category";

export class Task {
  id: string | undefined;
  name: string;
  description: string = "";
  startTime: Date;
  endTime: Date | null;
  pauseStart: Date | null;
  pauseTime: number;
  category: Category | null;

  constructor(name, description, startTime, category?) {
    this.name = name;
    this.description = description;
    this.startTime = startTime;
    if (category) {
      this.category = category;
    } else {
      this.category = null;
    }

    this.endTime = null;
    this.pauseStart = null;
    this.pauseTime = 0;
  }
}
